core/src/main/java/com/amplifyframework/util/UserAgent.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.util;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+/**
+ * A utility to construct a User-Agent header, to be sent with all network operations.
+ */
+public final class UserAgent {
+    private static String instance = null;
+
+    @SuppressWarnings("checkstyle:all") private UserAgent() {}
+
+    /**
+     * Gets a String to use as the value of a User-Agent header.
+     * @return A value for a User-Agent header.
+     */
+    @SuppressLint("SyntheticAccessor")
+    @NonNull
+    public static String string() {
+        if (instance == null) {
+            instance = new UserAgent.Builder()
+                .libraryName("amplify-android")
+                .libraryVersion("0.9")
```

Comment:
I am wondering if there is a way to read the version number from a version file so user-agent is updated automatically for future versions without requiring manual intervention.

Reply:
`BuildConfig.VERSION_NAME` is one possible candidate for this. It would mean we'd be updating the `build.gradle`'s central `versionName` field, instead of this `.java` file. Which would at least be a step in the right direction.

Perhaps there's a way to read the pom artifact version at runtime, too. Agree that it warrants a closer look. I'll do a quick investigation.

Reply:
^ Did the thing